apps/script/HoC2023ScriptFiles/README.md

```diff
@@ -1,25 +1,62 @@
+
+A simple flow diagram of how data/parameters move through the script would be:
+(constants.py + ai-inputs.json) -> HoC2023AiGenerateWeights.py -> Hoc2023AiHelperFunctions.py -> [LLM of Choice] -> HoC2023AiGenerateWeights.py -> Final Map Output
+
+## AWS Titan Setup and Quirks
+
+To be blunt, setting up AWS is challenging and the guidance/code samples change frequently. It is highly suggested to check their documentation at https://docs.aws.amazon.com/bedrock/latest/userguide/model-access.html, https://github.com/aws-samples/amazon-bedrock-workshop, and https://github.com/aws-samples/amazon-bedrock-workshop/blob/main/00_Intro/bedrock_boto3_setup.ipynb and do your best to follow along.
+
+Additionally, make sure to reach out to the Infrastructure team for best practices related to IAM permissions and role creation you will need to get Bedrock model access. Previously, we opted to utilize the recommended SageMaker service and role to streamline permissions issues but better options (for both cost and organization) might have since been determined.
+
+## Output Evaluation
+
+There isn't a strict numeric threshold for quality that the model must meet compared outside of ensuring that the final results meet the required format for DanceAI to reference later. However, we established the following desired characteristics for the distribution of model outputs as of 01/08/2024:
+
+1. Low standard deviation for each output category (e.g. roughly equal chance for any particular palette option to be chosen given infinite arbitrary input combinations)
+2. All output options are represented (e.g. if there are 29 total backgrounds, all 29 should have the potential of be chosen given infinite arbitrary input combinations)
+3. Sniff test - do the generated outputs match your intuitions on what that input should make?
+
+To aide in evaluating whether changes to the model/descriptive phrases are working towards or away from the above, see the Hoc2023GeneratorPerformanceEval.ipynb and associated hoc2023_prompt_performance_records.csv. Note that you might have to adjust file paths/names for HoC2023GeneratorPerformanceEval.ipynb to work on your local environment and that the evaluation criterion presented in the notebook have since been altered to the above.
+
+## Output Reference Lists
 
-Below are palettes, background effects, and foreground effects in the order that they appear in the dropdowns. When this file is run and the three cached data files are replaced in apps/static/dance/ai/model, please update the lists below with the output from this script.
+Below are palettes, background effects, and foreground effects in the order that they appear in the dropdowns as of 01/08/2024. When this file is run and the three cached data files are replaced in apps/static/dance/ai/model, please update the lists below with the output from this script.
 
 PALETTES  
```

Comment:
Instead, list all the places to find these in code. That will be the source of truth, and will keep the README.md from getting out of date.

Reply:
I originally included list and SGTM!